shopinvader_api_delivery_carrier/routers/delivery_carrier.py

```diff
@@ -0,0 +1,119 @@
+# Copyright 2023 ACSONE SA/NV
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+from typing import Annotated
+from uuid import UUID
+
+from fastapi import APIRouter, Depends, HTTPException
+
+from odoo import api, fields, models
+
+from odoo.addons.base.models.res_partner import Partner as ResPartner
+from odoo.addons.delivery.models.delivery_carrier import DeliveryCarrier
+from odoo.addons.fastapi.dependencies import (
+    authenticated_partner,
+    authenticated_partner_env,
+)
+
+from ..schemas import (
+    DeliveryCarrier as DeliveryCarrierSchema,
+    DeliveryCarrierSearch,
+    DeliveryCarrierWithPrice,
+)
+
+delivery_carrier_router = APIRouter(tags=["delivery_carriers"])
+
+
+# FIXME: not working if get method. But it should
+@delivery_carrier_router.post("/delivery_carriers")
+def search(
+    env: Annotated[api.Environment, Depends(authenticated_partner_env)],
+    partner: Annotated["ResPartner", Depends(authenticated_partner)],
+    data: DeliveryCarrierSearch,
+) -> list[DeliveryCarrierSchema]:
```

Comment:
@lmignon  I would need some help here:

This search method should be a GET, but it takes as input `data`, which is a `DeliveryCarrierSearch` (zipcode + country).
In unit tests, we must do a `test_client.post` because `test_client.get` doesn't allow content to be passed. I see you did the same in `shopinvader_api_wishlist`: https://github.com/shopinvader/odoo-shopinvader/blob/b64c0ca35a8a555eeb56f492ee9ce511b6783d61/shopinvader_api_wishlist/tests/test_wishlist.py#L34

But then, if this method is tagged as a `GET`, the tests return a 405 error,  I'm not sure to understand why. The only difference with your wishlist search GET method is that my route is authenticated...

Reply:
IMO we should handle such cases with GET requests with [query parameters](https://fastapi.tiangolo.com/tutorial/query-params/#query-parameters).

Otherwise we have to make it POST to allow for a request body but it makes no sense for a search to be a POST.

Reply:
I took my inspiration from https://github.com/shopinvader/odoo-shopinvader/blob/b64c0ca35a8a555eeb56f492ee9ce511b6783d61/shopinvader_api_wishlist/routers/wishlists.py#L37

The search is still a GET (even if it is tested with `client.post` in unit tests), and there it works.

But I'm not against using query parameters. @lmignon what do you think?

Reply:
I just discussed this with @lmignon, you can take inspiration from the [search in `/sales`](https://github.com/shopinvader/odoo-shopinvader/blob/0e68206ab924a2d155a1e79e6bcaacade66a6a03/shopinvader_api_sale/routers/sales.py#L32).
An empty `Depends()`  makes the argument a query parameter.

Reply:
Okay, right. But then I don't know how to write a succeeding unit test.
In https://github.com/shopinvader/odoo-shopinvader/blob/0e68206ab924a2d155a1e79e6bcaacade66a6a03/shopinvader_api_sale/tests/test_shopinvader_sale_api.py there is no test where `SaleSearch` is given as params. And when I try to add such a test, it also fails (405 error).
Hence the question is: how to reproduce a call to a GET with params in unit tests?
I just pushed the corrections to let you see the tests failing.

Reply:
It works, thanks!